src/compiler/checker.rs

```diff
@@ -2787,19 +3160,38 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
             self.report_error(&message, args);
         };
 
-        if self.type_checker.is_simple_type_related_to(
-            &*source,
-            &*target,
-            self.relation,
-            if report_errors {
-                Some(&report_error)
-            } else {
-                None
-            },
-        ) {
+        if false
+            || self.type_checker.is_simple_type_related_to(
+                &*source,
+                &*target,
+                self.relation,
+                if report_errors {
+                    Some(&report_error)
+                } else {
+                    None
+                },
+            )
+        {
             return Ternary::True;
         }
 
+        let is_performing_excess_property_checks = !intersection_state
+            .intersects(IntersectionState::Target)
+            && (self.type_checker.is_object_literal_type(source.clone())
+                && get_object_flags(&*source).intersects(ObjectFlags::FreshLiteral));
+        if is_performing_excess_property_checks {
+            if self.has_excess_properties(source.clone(), target.clone(), report_errors) {
+                if report_errors {
+                    self.report_relation_error(
+                        head_message,
+                        source,
+                        if false { original_target } else { target },
```

Comment:
It'd be totally redundant since if-statements are expressions, but I really wish Rust had the `? :` ternary operator.